macros/models/_sector/dex/uniswap_compatible_trades.sql

```diff
@@ -73,7 +74,11 @@ WITH dexs AS
         , CASE WHEN amount0 < INT256 '0' THEN f.token0 ELSE f.token1 END AS token_bought_address
         , CASE WHEN amount0 < INT256 '0' THEN f.token1 ELSE f.token0 END AS token_sold_address
         , t.contract_address as project_contract_address
-        , f.fee
+        {% if optional_columns %}
+            {% for optional_column in optional_columns %}
+            , {{ optional_column }}
+            {% endfor %}
+        {% endif %}
```

Comment:
Updated to a loop.

Reply:
can you guys help clarify my understanding of this?
- current `main` branch implementation has `f.fee` being pulled in CTE, but then left out of select downstream
- camelot decoded table(s) doesn't have this column, so it would fail on read
- the `{% if optional_columns %}` checks if the column exists, if yes, includes in CTE, if not, it leaves it out (i can confirm `dbt compile` returns no column)
  - is the if check essentially: try to pull it, if it will fail, then remove?

Reply:
Correct. For camelot optional_columns is passed as an empty list and if detects it as such and skips this section.

Reply:
No worries!